src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -4908,6 +4941,39 @@ instruct loadConNKlass(iRegNNoSp dst, immNKlass con)
   ins_pipe(ialu_imm);
 %}
 
+// Load Half Float Constant
+instruct loadConH(fRegF dst, immH con) %{
+  match(Set dst con);
+
+  ins_cost(LOAD_COST);
+  format %{
+    "flh $dst, [$constantaddress]\t# load from constant table: float=$con, #@loadConH"
+  %}
+
+  ins_encode %{
+    assert(UseZfh || UseZfhmin, "must");
+    __ flh(as_FloatRegister($dst$$reg), $constantaddress($con));
+    // TODO: add zfa instructions for half float, and optimize here.
+  %}
+
+  ins_pipe(fp_load_constant_s);
+%}
+
+instruct loadConH0(fRegF dst, immH0 con) %{
+  match(Set dst con);
+
+  ins_cost(XFER_COST);
+
+  format %{ "fmv.h.x $dst, zr\t# float, #@loadConH0" %}
+
+  ins_encode %{
+    assert(UseZfh || UseZfhmin, "must");
+    __ fmv_h_x(as_FloatRegister($dst$$reg), zr);
+  %}
+
+  ins_pipe(fp_load_constant_s);
+%}
```

Comment:
Can we put this two after `loadConD0`? Then we have a more consistent order for the three variants (F->D->FH) at each place.

Reply:
In fact, in my mind the order should be FH->F->D, considering the width changing from 16->32->64.

Reply:
That's also fine to me. Just expecting a consistent order for other definitions as well like `immHF0` -> `immF0` -> `immD0`.

Reply:
The orders in original code are inconsistent, e.g. for `operand` it's immD -> immF, for `instruct` it's loadConF -> loadConD, this patch just follow the previous local order.

And I think this inconsistency is not a problem, because when we read the code, as long as it maintains a certain order locally, then there is no problem, we don't have to enforce global consistency because it is difficult to do so and there doesn't seem to be much benefit.